src/amo/components/VPNPromoBanner/index.js

```diff
@@ -0,0 +1,138 @@
+/* @flow */
+import config from 'config';
+import deepEqual from 'deep-eql';
+import * as React from 'react';
+import { compose } from 'redux';
+import { withRouter } from 'react-router-dom';
+
+import Button from 'amo/components/Button';
+import tracking from 'amo/tracking';
+import translate from 'amo/i18n/translate';
+import { makeQueryStringWithUTM } from 'amo/utils';
+import type { I18nType } from 'amo/types/i18n';
+import type { ReactRouterLocationType } from 'amo/types/router';
+
+import './styles.scss';
+import vpnLogo from './img/mozilla-vpn.svg';
+import dismissX from './img/x-close-black.svg';
+
+export const VPN_PROMO_CAMPAIGN = 'amo-vpn-promo';
+export const VPN_PROMO_CATEGORY = 'VPN Promo Banner';
+export const VPN_PROMO_CLICK_ACTION = 'vpn-promo-banner-click';
+export const VPN_PROMO_DISMISS_ACTION = 'vpn-promo-banner-dismiss';
+export const VPN_PROMO_IMPRESSION_ACTION = 'vpn-promo-banner-impression';
+export const VPN_URL = 'https://www.mozilla.org/products/vpn';
+
+export type Props = {||};
+
+export type DeafultProps = {|
+  _config: typeof config,
+  _tracking: typeof tracking,
+|};
+
+type InternalProps = {|
+  ...Props,
+  ...DeafultProps,
+  i18n: I18nType,
+  location: ReactRouterLocationType,
+|};
+
+export class VPNPromoBannerBase extends React.Component<InternalProps> {
+  static defaultProps: DeafultProps = {
+    _config: config,
+    _tracking: tracking,
+  };
+
+  onButtonClick: () => void = () => {
+    this.props._tracking.sendEvent({
+      action: VPN_PROMO_CLICK_ACTION,
+      category: VPN_PROMO_CATEGORY,
+    });
+  };
+
+  onDismiss: () => void = () => {
+    this.props._tracking.sendEvent({
+      action: VPN_PROMO_DISMISS_ACTION,
+      category: VPN_PROMO_CATEGORY,
+    });
+  };
+
+  onImpression: () => void = () => {
+    const { _config, _tracking } = this.props;
+
+    if (_config.get('enableFeatureVPNPromo')) {
+      _tracking.sendEvent({
+        action: VPN_PROMO_IMPRESSION_ACTION,
+        category: VPN_PROMO_CATEGORY,
+      });
+    }
+  };
+
+  componentDidMount() {
+    this.onImpression();
+  }
+
+  componentDidUpdate(prevProps: InternalProps) {
+    if (!deepEqual(this.props.location, prevProps.location)) {
+      this.onImpression();
+    }
+  }
+
+  render(): null | React.Node {
+    const { _config, i18n } = this.props;
+    const headline = i18n.gettext(
+      'Introductory offer ends soon: $4.99/month for Mozilla VPN',
```

Comment:
We should probably not let localizers decide on the pricing so we need to extract the price (+ currency probably) and, depending on the locale, choose the right price, fallbacking to the `en-US` price/currency unless we completely hide this banner for unsupported locales.

Reply:
The code for monitor doesn't do this, and in fact the current copy (from which I should update this), doesn't include a price. Monitor simply uses translated strings for each of these pieces of text (e.g., https://github.com/mozilla/blurts-server/blob/main/locales/de/app.ftl#L577-L581). We *are* only going to display the banner for supported regions (there's a separate issue for that), and, for the purposes of the experiment, I don't think it's worth doing anything fancier than what monitor is currently doing. I will update our strings to match the ones they are currently using.

Reply:
Right, if the copy does not include any price, then it is different.